packages/nextclade/src/analyze/find_relative_mutations.rs

```diff
@@ -0,0 +1,137 @@
+  ref_nodes
+    .iter()
+    .map(|&ref_node| -> Result<_, Report> {
+      let node = graph
+        .iter_nodes()
+        .find(|node| node.payload().name == ref_node.name)
+        .ok_or_else(|| eyre!("Unable to find reference node on the tree: '{}'", &ref_node.name))?;
+
+      let muts = find_private_nuc_mutations(
+        node.payload(),
+        substitutions,
+        deletions,
+        missing,
+        alignment_range,
+        ref_seq,
+        non_acgtns,
+        virus_properties,
+      );
+
+      Ok(RelativeNucMutations {
+        ref_node: ref_node.to_owned(),
+        muts,
+      })
+    })
+    .collect::<Result<Vec<RelativeNucMutations>, Report>>()
+    .wrap_err("When calling nucleotide mutations relative to a reference node")
+}
+
+#[derive(Clone, Serialize, Deserialize, schemars::JsonSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct RelativeAaMutations {
+  ref_node: AuspiceRefNode,
+  muts: BTreeMap<String, PrivateAaMutations>,
+}
+
+pub fn find_relative_aa_mutations(
+  graph: &AuspiceGraph,
+  clade: &str,
+  clade_node_attrs: &BTreeMap<String, String>,
+  aa_substitutions: &[AaSub],
+  aa_deletions: &[AaDel],
+  aa_unknowns: &[CdsAaRange],
+  aa_unsequenced_ranges: &BTreeMap<String, Vec<AaRefRange>>,
+  ref_peptides: &Translation,
+  gene_map: &GeneMap,
+) -> Result<Vec<RelativeAaMutations>, Report> {
+  let ref_nodes = filter_ref_nodes(graph, clade, clade_node_attrs);
+
+  ref_nodes
+    .iter()
+    .map(|&ref_node| -> Result<_, Report> {
+      let node = graph
+        .iter_nodes()
+        .find(|node| node.payload().name == ref_node.name)
+        .ok_or_else(|| eyre!("Unable to find reference node on the tree: '{}'", &ref_node.name))?;
+
+      let muts = find_private_aa_mutations(
+        node.payload(),
+        aa_substitutions,
+        aa_deletions,
+        aa_unknowns,
+        aa_unsequenced_ranges,
+        ref_peptides,
+        gene_map,
+      );
+
+      Ok(RelativeAaMutations {
+        ref_node: ref_node.to_owned(),
+        muts,
+      })
+    })
+    .collect::<Result<Vec<RelativeAaMutations>, Report>>()
+    .wrap_err("When calling amino acid mutations relative to a reference node")
+}
+
+/// Take only ref nodes which are relevant for this sample, according to the node's include list
+pub fn filter_ref_nodes<'a>(
+  graph: &'a AuspiceGraph,
+  clade: &str,
+  clade_node_attrs: &BTreeMap<String, String>,
+) -> Vec<&'a AuspiceRefNode> {
+  graph
+    .data
+    .meta
+    .extensions
+    .nextclade
+    .reference_nodes
+    .iter()
+    .filter(|node| {
+      // For each attribute key in includes, check that the attribute value of this sample match
+      // at least one item in the include list
+      node.include.iter().all(|(key, includes)| {
+        let curr_value = if key == "clade" { clade } else { &clade_node_attrs[key] };
+        includes.iter().any(|include_value| include_value == curr_value) // TODO: consider regex match rather than equality
+      })
+    })
```

Comment:
This is the logic for constraining the mutations calculation by clades and clade-like attributes. If `include` field is present, then we lookup the constrained attribute of the query sample and only consider this node if the query attribute's value is matching any of the values in the include list.

For example, if config has node of interest which is only relevant for clades 23A and 23B:

```json
{
  include: { clade: ["23A", "23B"] }
}
```

then mutations relative to this node will be calculated only for query samples of clade 23A and 23B.


Same for pango lineages:

```json
{
  include: { Nextclade_pango: ["A.1.2.3", "A.1.2.3.4"] }
}
```

It is up for discussion how multiple keys in the `include` object should be handled - using boolean OR and AND.